### PR TITLE
Prepend Class Auto Loader

### PR DESCRIPTION
**Pull request by Fenikkusu**
The auto loader class currently appends itself when calling spl_autoloader_register. This can cause the Phalcon loader to not be the primary loader when working with other registered autoloaders. The best example is Composer. Composer, by default, prepends its autoloader, making it the first to run.

The applied changes makes use of the same concept to prepend the Phalcon loader. This, in combination with the recent 'registerFiles' method allows Phalcon to always be the primary loader.

For backwards compatability, the loader defaults to appending still, but can be converted to prepending by calling setPrepend(TRUE)

$loader = new \Phalcon\Loader();
$loader->registerNamespaces(['SomeNamespace' => 'Some\Namespace'])
->setPrepend(TRUE)
->register();

### CODE REVIEW

File: tests/_data/vendor/Example/Test/YetAnotherTest.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+    /**
+     * Created by PhpStorm.
+     * User: fenikkusu
+     * Date: 8/5/16
+     * Time: 10:28 PM
+     */
+    class YetAnotherTest {
+
```
Comment (sergeyklay) [inline tests/_data/vendor/Example/Test/YetAnotherTest.php]: Could you please remove these useless comments

File: tests/_data/vendor/Example/Test/AnotherTest.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+    /**
+     * Created by PhpStorm.
+     * User: fenikkusu
+     * Date: 8/5/16
+     * Time: 10:28 PM
+     */
+    class AnotherTest {
```
Comment (sergeyklay) [inline tests/_data/vendor/Example/Test/AnotherTest.php]: the same

**Comment by Fenikkusu** [comment]
I've removed the auto-generated comments.

**Comment by Fenikkusu** [comment]
I was curious as to why this was closed with no merger. Found the other PR doing the same thing in slightly different manner. Any particular reason that one was used? Given this had been waiting for 7 months prior, I'm slightly confused.

**Comment by sergeyklay** [comment]
After deleting 3.0.x branch this issues was closed automatically because your PR is still directed to it.

Could you please submit your PR to the 3.2.x branch?

**Comment by Fenikkusu** [comment]
I can, but didn't the PR from a month ago (#12580) render this null/void?

**Pull request closed without merging**